More detailed hydra test results

AsparagusEduardo opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Added some more details to the test results.

![image](https://user-images.githubusercontent.com/2904965/220514476-45c4a6d3-ed50-4ded-b39c-100c599ed9f6.png)

`KNOWN_FAILING` and `FAILED` results don't show if there are none.

![image](https://user-images.githubusercontent.com/2904965/220514786-934bc665-75bb-4540-8d6a-bd17520b0546.png)


## **Discord contact info**
AsparagusEduardo#6051

DizzyEggg commented:
What do you think about adding an additional field for tests which failed due to assumption being false? Perhaps ASSUMPTION_FAILED or sth similar?

mrgriffin commented on tools/mgba-rom-test-hydra/main.c:
```diff
@@ -9,7 +9,11 @@
  * COMMANDS
  * N: Sets the test name to the remainder of the line.
  * R: Sets the result to the remainder of the line, and flushes any
- *    output buffered since the previous R. */
+ *    output buffered since the previous R.
+ *
+ * //Missing P, F and K documentation, please tell me what to put here lol
+ * 
+ */
```
Oops, mb!

I suppose something like:
```
P/K/F: Sets the result to the remaining of the line, flushes any output since the previous P/K/F, and for P/K increments the number of passes/known fails.
```

AsparagusEduardo commented:
> What do you think about adding an additional field for tests which failed due to assumption being false? Perhaps ASSUMPTION_FAILED or sth similar?

Sure thing :)

AsparagusEduardo commented:
Added `ASSUMPTIONS_FAIL` to the result summary.
Also changed `SKIP` to `ASSUMPTION_FAIL` and made it yellow.

![image](https://user-images.githubusercontent.com/2904965/220718286-b903a885-d78f-4404-abc5-a6e38d8b76ec.png)

mrgriffin commented on test/test_runner.c:
```diff
@@ -187,15 +187,22 @@ void CB2_TestRunner(void)
                 }
                 break;
             case TEST_RESULT_PASS: result = "PASS"; break;
-            case TEST_RESULT_SKIP: result = "SKIP"; break;
+            case TEST_RESULT_SKIP:
+                result = "ASSUMPTION_FAIL";
+                color = "\e[33m";
+                break;
             case TEST_RESULT_INVALID: result = "INVALID"; break;
             case TEST_RESULT_ERROR: result = "ERROR"; break;
             case TEST_RESULT_TIMEOUT: result = "TIMEOUT"; break;
             default: result = "UNKNOWN"; break;
```
At this point, I'd probably just format all the cases on multiple lines :)

SBird1337 replied:
are there other reasons tests might be skipped? For readability I would probably rephrase it to `TEST_RESULT_ASSUMPTION_FAIL` or something along those lines.

AsparagusEduardo replied:
Yeah, I didn't find other causes for skipping, so it would make sense.

AsparagusEduardo commented:
Test

AsparagusEduardo commented:
Ready for re-review
